cellprofiler/modules/plugins/imagetemplate.py

```diff
@@ -49,125 +128,146 @@
-            choose any image that is made available by a prior module.
-            <br>
-            <b>ImageTemplate</b> will do something to this image.
-            """)
+        # reST help that gets displayed when the user presses the
+        # help button to the right of the edit box.
         #
-        # The ImageNameProvider makes the image available to subsequent
-        # modules.
+        # The superclass defines some generic help test. You can add
+        # module-specific help text by modifying the setting's "doc"
+        # string.
         #
-        self.output_image_name = cps.ImageNameProvider(
-                "Output image name:",
-                # The second parameter holds a suggested name for the image.
-                "OutputImage",
-                doc="""This is the image resulting from the operation.""")
+        self.x_name.doc = """\
+This is the image that the module operates on. You can choose any image
+that is made available by a prior module.
+
+**ImageTemplate** will do something to this image.
+"""
+
         #
         # Here's a choice box - the user gets a drop-down list of what
         # can be done.
         #
-        self.gradient_choice = cps.Choice(
-                "Gradient choice:",
-                # The choice takes a list of possibilities. The first one
-                # is the default - the one the user will typically choose.
-                [GRADIENT_MAGNITUDE, GRADIENT_DIRECTION_X, GRADIENT_DIRECTION_Y],
-                #
-                # Here, in the documentation, we do a little trick so that
-                # we use the actual text that's displayed in the documentation.
-                #
-                # %(GRADIENT_MAGNITUDE)s will get changed into "Gradient magnitude"
-                # etc. Python will look in globals() for the "GRADIENT_" names
-                # and paste them in where it sees %(GRADIENT_...)s
-                #
-                # The <ul> and <li> tags make a neat bullet-point list in the docs
-                #
-                doc="""Choose what to calculate:
-            <ul>
-            <li><i>%(GRADIENT_MAGNITUDE)s</i> to calculate the
-            magnitude of the gradient at each pixel.</li>
-            <li><i>%(GRADIENT_DIRECTION_X)s</i> to get the relative contribution
-            of the gradient in the X direction (.5 = no contribution,
-            0 to .5 = decreasing with increasing X, .5 to 1 = increasing
-            with increasing X).</li>
-            <li><i>%(GRADIENT_DIRECTION_Y)s</i> to get the relative
-            contribution of the gradient in the Y direction.</li></ul>
-            """ % globals()
+        self.gradient_choice = cellprofiler.setting.Choice(
+            text="Gradient choice:",
+            # The choice takes a list of possibilities. The first one
+            # is the default - the one the user will typically choose.
+            choices=[GRADIENT_MAGNITUDE, GRADIENT_DIRECTION_X, GRADIENT_DIRECTION_Y],
```

Comment:
It might also be good to add a `value` keyword here to showcase default options.